app/src/main/kotlin/com/wire/android/ui/userprofile/common/UserProfileInfo.kt

```diff
@@ -75,76 +75,76 @@ fun UserProfileInfo(
-                    .constrainAs(editButton) {
-                        top.linkTo(userDescription.top)
-                        bottom.linkTo(userDescription.bottom)
-                        end.linkTo(userDescription.end)
-                    },
-                onClick = editableState.onEditClick,
-                content = Icons.Filled.Edit.Icon()
-            )
-        }
+            val (userDescription, editButton, teamDescription) = createRefs()
 
-        if (teamName != null) {
-            Text(
+            Column(
+                horizontalAlignment = CenterHorizontally,
                 modifier = Modifier
-                    .padding(top = dimensions().spacing8x)
-                    .padding(horizontal = dimensions().spacing16x)
-                    .constrainAs(teamDescription) {
-                        top.linkTo(userDescription.bottom)
+                    .padding(horizontal = dimensions().spacing64x)
+                    .wrapContentSize()
+                    .constrainAs(userDescription) {
+                        top.linkTo(parent.top)
+                        bottom.linkTo(parent.bottom)
                         start.linkTo(parent.start)
                         end.linkTo(parent.end)
-                    },
-                text = teamName,
-                maxLines = 1,
-                overflow = TextOverflow.Ellipsis,
-                style = MaterialTheme.wireTypography.label01,
-                color = MaterialTheme.colorScheme.onBackground,
-            )
+                    }
+            ) {
+                Text(
+                    text = fullName,
+                    overflow = TextOverflow.Ellipsis,
+                    maxLines = 1,
+                    style = MaterialTheme.wireTypography.title02,
+                    color = MaterialTheme.colorScheme.onBackground,
+                )
+                Text(
+                    text = "@$userName",
+                    overflow = TextOverflow.Ellipsis,
+                    style = MaterialTheme.wireTypography.body02,
+                    maxLines = 1,
+                    color = MaterialTheme.wireColorScheme.labelText,
+                )
+                if (membership.hasLabel()) {
+                    Spacer(Modifier.height(dimensions().spacing8x))
+                    MembershipQualifierLabel(membership)
+                }
+            }
+
+            if (editableState is EditableState.IsEditable) {
+                IconButton(
+                    modifier = Modifier
+                        .padding(start = dimensions().spacing16x)
+                        .constrainAs(editButton) {
+                            top.linkTo(userDescription.top)
+                            bottom.linkTo(userDescription.bottom)
+                            end.linkTo(userDescription.end)
+                        },
+                    onClick = editableState.onEditClick,
+                    content = Icons.Filled.Edit.Icon()
+                )
```

Comment:
Same here about extracting.

Without building the app, I **think** this is about editing the member? But I'm not sure exactly what kind of editing that would be. Is it editing the membership? Like making Admin or removing from group?

```suggestion
                ManageMemberButton(editableState.onEditClick)
```